django/core/files/storage.py

```diff
@@ -324,6 +339,25 @@ def created_time(self, name):
     def modified_time(self, name):
         return datetime.fromtimestamp(os.path.getmtime(self.path(name)))
 
+    def _do_safe_operation(self, operation, src_path, dst_path):
+        try:
+            operation(src_path, dst_path)
+        except OSError as e:
+            if e.errno == errno.EEXIST:
+                raise ValueError('Destination already exists')
```

Comment:
On the other hand, if I go with option 3 and have clients handle failure modes differently depending on backend, then that defeats the whole point of having these methods in the Storage API.

They could just write their own copy/move functions in their own apps that knows how to deal with their configured backend.

I'll close this and mark the ticket as wontfix.